postgres-testing/src/test/java/org/folio/postgres/testing/PostgresTesterContainerTest.java

```diff
@@ -48,4 +58,70 @@ public void testGetHost() {
   public void testGetPort() {
     new PostgresTesterContainer().getPort();
   }
+
+  @Test(expected = IllegalStateException.class)
+  public void testGetReadHost() {
+    new PostgresTesterContainer().getReadHost();
+  }
+
+  @Test(expected = IllegalStateException.class)
+  public void testGetReadPort() { new PostgresTesterContainer().getReadPort(); }
+
+  @Test(expected = IllegalStateException.class)
+  public void testGetNetwork() { new PostgresTesterContainer().getNetwork(); }
+
+  @Test
+  public void testReadWrite() throws SQLException, PostgresTesterStartException {
+    PostgresTester tester = new PostgresTesterContainer();
+    String user = "user";
+    String db = "db";
+    String pass = "pass";
+    tester.start(db, user, pass);
+
+    // Connect to the read-write host and create some data.
+    int port = tester.getPort();
+    int readOnlyPort = tester.getReadPort();
+    String host = tester.getHost();
+    String readOnlyHost = tester.getReadHost();
+
+    String connString = String.format("jdbc:postgresql://%s:%d/%s", host, port, db);
+    String connStringReadOnly = String.format("jdbc:postgresql://%s:%d/%s", readOnlyHost, readOnlyPort, db);
+
+    try (Connection conn = DriverManager.getConnection(connString, user, pass);
+         Connection readOnlyConn = DriverManager.getConnection(connStringReadOnly, user, pass);
+         Statement stmt = conn.createStatement();
+         Statement readOnlyStmt = readOnlyConn.createStatement()) {
+
+      // Check that streaming replication is set up correctly.
+      ResultSet state = stmt.executeQuery("SELECT state FROM pg_stat_replication;");
+      Assert.assertTrue(state.next());
+      Assert.assertEquals("streaming", state.getString("state"));
+
+      // The standby should not accept writes of any kind since it is read-only.
+      Exception exception = Assert.assertThrows(PSQLException.class, () -> {
+        readOnlyStmt.executeUpdate("CREATE TABLE crew (id SERIAL PRIMARY KEY, name VARCHAR(50));");
+      });
+      Assert.assertEquals("ERROR: cannot execute CREATE TABLE in a read-only transaction", exception.getMessage());
+
+      // Should we test synchronous commit? The default is true.
+      boolean testSynchronousCommit = System.getProperty(PostgresTesterContainer.POSTGRES_ASYNC_COMMIT) == null;
+      if (!testSynchronousCommit) {
+        return;
+      }
+
+      // The first writes to the replicated cluster requires remote_apply. Subsequent ones don't.
+      stmt.executeUpdate("BEGIN; SET LOCAL synchronous_commit = remote_apply; CREATE TABLE crew (id SERIAL PRIMARY KEY, name VARCHAR(50)); COMMIT;");
```

Comment:
Well in your branch you have removed this and it appears to work without it, probably because you are setting the synchronous_standby_names on init of the primary. Fine and good.